plugins/processors/template/template.go

```diff
@@ -0,0 +1,38 @@
+package template
+
+import (
+	"github.com/influxdata/telegraf"
+	"strings"
+	"text/template"
+)
+
+type TemplateProcessor struct {
+	Tag      string `toml:"tag"`
+	Template string `toml:"template"`
+}
+
+func (r *TemplateProcessor) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	// create template
+	tmpl, err := template.New("test").Parse(r.Template)
+	if err != nil {
+		panic(err)
+	}
+
+	// for each metric in "in" array
+	for _, metric := range in {
+		var b strings.Builder
+		//newM := TemplateMetric{metric}
```

Comment:
When you pass this into Execute, make sure to pass a pointer.  Either `&TemplateMetric{metric}` or `&newM` below.